src/deepsparse/loggers/function_logger.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Implementation of the Function Logger
+"""
+from typing import Any, Callable
+
+from deepsparse.loggers import BaseLogger, MetricCategories
+from deepsparse.loggers.helpers import match, possibly_extract_value
+
+
+__all__ = ["FunctionLogger"]
+
+
+class FunctionLogger(BaseLogger):
+    """
+    DeepSparse logger that applies functions to raw,
+    logged values (collected in the log() method)
+    according to FunctionLogger's attributes
+
+    :param logger: A child DeepSparse Logger object
+    :param identifier: The string that needs to match the
```

Comment:
For now, I am tying the "self.identifier" (something that would be fed to the FunctionLogger by the MatchingLogger in Corey's concept) to the FunctionLogger itself. Something we can easily change once we understand the UI more. Even though it will increase the number FunctionLoggers overall, it simplifies the hierarchy of loggers (Leafloggers <-- FunctionLoggers <-- MultiLogger)

Reply:
let's come back to this - still in favor of limiting the number of "Loggers" instantiated, additionally in practice this would make the heiarchy `LeafLoggers <-- MultiLogger <-- FunctionLoggers <-- MultiLogger <= SubProcLogger` for many use cases

Reply:
I agree.